src/libmdb/data.c

```diff
@@ -1046,9 +1052,6 @@ char *mdb_col_to_string(MdbHandle *mdb, void *buf, int start, int datatype, int
 					size, text, mdb->bind_size);
 			}
 		break;
-		case MDB_DATETIME:
-			text = mdb_date_to_string(mdb, mdb->date_fmt, buf, start);
-		break;
```

Comment:
from what I can tell, there isn't a path to get here in either `data.c` or `odbc.c`, and this doesn't handle the difference between the date_fmt and short_date_fmt.